asmcomp/arm/emit.mlp

```diff
@@ -609,6 +609,7 @@ let emit_instr i =
         let instr = name_for_int_operation op in
         `	{emit_string instr}	{emit_reg i.res.(0)}, {emit_reg i.arg.(0)}, #{emit_int n}\n`; 1
     | Lop(Iabsf | Inegf as op) when !fpu = Soft ->
+        assert (i.res.(0).loc = i.arg.(0).loc);
```

Comment:
I suggest we add a similar assertion (but an inequality) for `Iintop Imul` and `Ispecific Imuladd` when `!arch < ARMv6`.

Reply:
That's not necessary because the assembler will give an error if the architectural constraint is not met. (Like it did in the original MPR.)  And the error message produced by the assembler is more legible than "assertion failed".  

For the other two cases (those found by @mshinwell) the assertion is much more valuable because the error cannot be seen on the asm output.